octodns/provider/cloudflare.py

```diff
@@ -206,6 +248,20 @@ def _include_change(self, change):
             new['ttl'] = max(120, new['ttl'])
             if new == existing:
                 return False
+
+        # If this is a record to enable Cloudflare CDN don't update as
+        # we don't know the original values.
+        if (hasattr(change.new, '_type') and
+            (change.new._type == 'CNAME' or
+             change.new._type == 'ALIAS') and
+                change.new.value.endswith('.cdn.cloudflare.net.')):
+            return False
+        if (hasattr(change.existing, '_type') and
+            (change.existing._type == 'CNAME' or
+             change.existing._type == 'ALIAS') and
+                change.existing.value.endswith('.cdn.cloudflare.net.')):
+            return False
```

Comment:
Sorry didn't really catch this in the last round. I assume the `hasattr` stuff here is around trying to handle changes that don't have `new` or `existing` by testing to see if `_type` is there, which it won't be on `None`?

If so it can be simplified to something like:

```python
if (change.record._type in ('ALIAS', 'CNAME') and
    change.record.value.endswith('.cdn.cloudflare.net.')):
    return False
```

The `.record` property was added to the `Change` object to make cases like this easier to deal with by always returning a record from the change object regardless of what type of change it is.

Otherwise this looks gtg. Soon as that comes through I'll :shipit: